handlers/engine_test.go

```diff
@@ -0,0 +1,24 @@
+package handlers
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	gock "gopkg.in/h2non/gock.v1"
+)
+
+func TestEngineError(t *testing.T) {
+	defer gock.Off()
+	GockStatus("status request").ReplyError(fmt.Errorf("Generated error"))
+	rr := serveURL("output=raw")
+	assert.Equal(t, 500, rr.Code)
```

Comment:
use `require.Equal` over `assert.Equal` this will stop test execution, and then you don't have to worry about future nil checks which can mask test failures